panel/layout/modal.py

```diff
@@ -0,0 +1,187 @@
+"""The `Modal` is an element that displays in front of and deactivates all other page content."""
+import param
+
+from ..reactive import ReactiveHTML
+from .base import NamedListLike
+
+# See https://a11y-dialog.netlify.app/
+JS_FILE = "https://cdn.jsdelivr.net/npm/a11y-dialog@7/dist/a11y-dialog.min.js"
+
+STYLE = """
+.dialog-container,
+.dialog-overlay {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+}
+.dialog-container {
+  z-index: 100002;
+  display: flex;
+}
+.dialog-container[aria-hidden='true'] {
+  display: none;
+}
+.dialog-overlay {
+  z-index: 100001;
+  background-color: rgb(43 46 56 / 0.9);
+}
+.dialog-content {
+  margin: auto;
+  z-index: 100002;
+  position: relative;
+  background-color: white;
+  border-radius: 2px;
+  padding: 10px;
+  padding-bottom: 20px;
+}
+fast-design-system-provider .dialog-content {
+  background-color: var(--background-color);
+  border-radius:  calc(var(--corner-radius) * 1px);
+}
+@keyframes fade-in {
+  from {
+    opacity: 0;
+  }
+}
+@keyframes slide-up {
+  from {
+    transform: translateY(10%);
+  }
+}
+.dialog-overlay {
+  animation: fade-in 200ms both;
+}
+.dialog-content {
+  animation: fade-in 400ms 200ms both, slide-up 400ms 200ms both;
+}
+@media (prefers-reduced-motion: reduce) {
+  .dialog-overlay,
+  .dialog-content {
+    animation: none;
+  }
+}
+.pnx-dialog-close {
+  position: absolute;
+  top: 0.5em;
+  right: 0.5em;
+  border: 0;
+  padding: 0.25em;
+  background-color: transparent;
+  font-size: 1.5em;
+  width: 1.5em;
+  height: 1.5em;
+  text-align: center;
+  cursor: pointer;
+  transition: 0.15s;
+  border-radius: 50%;
+  z-index: 10003;
+}
+fast-design-system-provider .pnx-dialog-close {
+  color:  var(--neutral-foreground-rest);
+}
+.pnx-dialog-close:hover {
+  background-color: rgb(50 50 0 / 0.15);;
+}
+fast-design-system-provider .pnx-dialog-close:hover {
+  background-color: var(--neutral-fill-hover);
+}
+.lm-Widget.p-Widget.lm-TabBar.p-TabBar.lm-DockPanel-tabBar.jp-Activity {
+  z-index: -1;
+}
+"""
+
+class Modal(ReactiveHTML, NamedListLike):  # pylint: disable=too-many-ancestors
```

Comment:
- [ ] Should figure out if its possible to add parameter to disable the closing of the modal when clicking the back drop.